src/object/README.md

```diff
@@ -34,11 +34,66 @@ In addition user can register customized object class by
 object class is stored as container metadata; it is valid in the lifetime of the
 container.
 
-The object class ID is embedded in object ID. By `daos_obj_generate_id()` user
+The object class ID is embedded in object ID. By `daos_obj_generate_oid()` user
 can generate an object ID for the specific object class ID. DAOS uses this class
 ID to find the corresponding object class, and then distribute and protect
 object data based on algorithm descriptions of this class.
 
+Users can select the object class manually when generating the oid from the list of all object
+classes in [/src/include/daos_obj_class.h]. However manually selecting the object class is not
+encouraged for regular users and should be done by advanced users only who understand all the
+different object classes and the redundancy factor of the container. For most users, passing an
+OC_UNKNOWN (0) object class to `daos_obj_generate_oid()` would allow DAOS to automatically select an
+object class based on the container properties where that object is being accessed such as the
+redundancy factor (RF), the number of domain (server engines) of the pool, and on the type of object
+being accessed (determined by the feats flag).
+
+The following details how the object class is chosen when no default or hints are provided:
+- RF:0
+  - Array, Byte Array, Flat KV object: OC_SX
+  - no feats type: OC_S1
+- RF:1
+  - Array, Byte Array:
+    - domain_nr >= 10 : OC_EC_8P1GX
```

Comment:
Do we need to mention the higher range of EC class ?
- domain_nr >= 18 : OC_EC_16P1GX

Reply:
this is not implemented currently.. as per wangdi's advice, we check up to 10 and use 8Pn if more.. if we need to add that then we probably need a different ticket to update the selection logic and this doc